sdk/storage/azblob/blob/client.go

```diff
@@ -525,7 +521,7 @@ func (b *Client) downloadFile(ctx context.Context, writer io.Writer, o downloadO
 		OperationName: "downloadBlobToWriterAt",
 		TransferSize:  count,
 		ChunkSize:     o.BlockSize,
-		NumChunks:     numChunks,
+		NumChunks:     uint16(numChunks),
```

Comment:
We are converting the type of this variable twice. In the first loop to float64 and then here to uint16. While assigning the value itself if we convert to uint16 then this can be avoided. Older code which is removed was doing the same I see.

Reply:
Updated to reduce type conversion.